### PR TITLE
feat/filter worker by name

### PR DESCRIPTION
**Pull request by WollasRomeiro**

### CODE REVIEW

File: src/api/worker/worker.service.ts [hunk 1]
```diff
@@ -28,9 +29,14 @@ export class WorkerService {
     }
   }
 
-  async findAll(paginationOptions: IPaginationOptions): Promise<Pagination<SelectWorkerDto>> {
+  async findAll(workerFilter: WorkerFilter, paginationOptions: IPaginationOptions): Promise<Pagination<SelectWorkerDto>> {
+    const { name } = workerFilter;
     const query = this.repository.createQueryBuilder('worker').select().orderBy('worker.id', 'DESC');
 
+    if (name) {
+      query.andWhere('worker.name = :name', { name: name });
```
Comment (juniorpb) [inline src/api/worker/worker.service.ts]:
```sql
'worker.name ILIKE :name', { name: `${name}% }
```

File: src/api/worker/dto/worker-filter.dto.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class WorkerFilter {
+  @ApiProperty({ name: 'name', type: Number, required: false, example: '1' })
```
Comment (juniorpb) [inline src/api/worker/dto/worker-filter.dto.ts]: type: Number ?  example: '1' ? 
copy and past... rsrs

Comment (WollasRomeiro) [reply src/api/worker/dto/worker-filter.dto.ts]: Vacileition my

File: src/api/worker/worker.service.ts (hunk 1 above)
Comment (WollasRomeiro) [reply src/api/worker/worker.service.ts]: Done

**Pull request merged**